zlib.scm

```diff
@@ -136,11 +139,16 @@
            (unless eof? ; free up memory
              (inflate-end stream)))))))
 
-(define deflate-init (foreign-lambda int "deflateInit" z-stream int))
+(define deflate-init (foreign-lambda int "deflateInit2" z-stream int int int int int))
 (define deflate (foreign-lambda int "deflate" z-stream int))
 (define deflate-end (foreign-lambda void "deflateEnd" z-stream))
 
-(define (open-zlib-compressed-output-port #!optional (port (current-output-port)))
+(define (open-zlib-compressed-output-port #!optional (port (current-output-port))
+                                          #!key (level -1)
+                                                (method Z_DEFLATED)
+                                                (window-bits 15)
+                                                (mem-level 8)
```

Comment:
Likewise I'd use `DEF_MEM_LEVEL`.

Reply:
Used the constant here but I'm not sure about capping it at 8. In zconf.h I see:

```c
/* Maximum value for memLevel in deflateInit2 */
#ifndef MAX_MEM_LEVEL
#  ifdef MAXSEG_64K
#    define MAX_MEM_LEVEL 8
#  else
#    define MAX_MEM_LEVEL 9
#  endif
#endif
```

Reply:
Sorry, I meant capping the value of `mem-level` to the value of `MAX_MEM_LEVEL`. While 8 is a conservative value for that, it can be higher than that.

Reply:
Understood, added